core/blockchain.go

```diff
@@ -347,6 +347,12 @@ func NewBlockChain(db ethdb.Database, cacheConfig *CacheConfig, genesis *Genesis
 	// Make sure the state associated with the block is available, or log out
 	// if there is no available state, waiting for state sync.
 	head := bc.CurrentBlock()
+	// Make sure Cel2 chain has migrated data from Celo1
+	if chainConfig.Cel2Time != nil && !chainConfig.IsCel2(head.Time) {
+		log.Error("Migrated data for Cel2 is missing, please ensure the migrated data from Celo1 can be properly loaded before starting the blockchain",
+			"current number", head.Number, "current hash", head.Hash())
+		return nil, fmt.Errorf("missing migrated data")
+	}
```

Comment:
Hey @Kourin1996, you will need to take into account the syncmode as well, this condition is okay if the node are snap syncing, but should be flagged if the node is full syncing.

Reply:
I need to check that snap sync actually occurs from block #1.

Reply:
@piersy I confirmed that snap sync mode could advance blocks even without migrated data. I added new constraints for restricting sync mode, however I needed to move this check to backend.
5c3e2be2d0fbe1dc9b6c8c0d2c2d112b5edee433